components/edit_post_modal/edit_post_modal.tsx

```diff
@@ -4,65 +4,117 @@
 
 import DeletePostModal from 'components/delete_post_modal';
 import EmojiPickerOverlay from 'components/emoji_picker/emoji_picker_overlay.jsx';
 import EmojiIcon from 'components/widgets/icons/emoji_icon';
 import Textbox from 'components/textbox';
+import TextboxClass from 'components/textbox/textbox';
 import TextboxLinks from 'components/textbox/textbox_links';
+import TextboxLinksClass from 'components/textbox/textbox_links/textbox_links';
+import {CustomEmoji, Emoji, SystemEmoji} from 'mattermost-redux/types/emojis';
 
 const KeyCodes = Constants.KeyCodes;
 const TOP_OFFSET = 0;
 const RIGHT_OFFSET = 10;
 
-class EditPostModal extends React.PureComponent {
-    static propTypes = {
-        canEditPost: PropTypes.bool,
-        canDeletePost: PropTypes.bool,
-        codeBlockOnCtrlEnter: PropTypes.bool,
-        ctrlSend: PropTypes.bool,
-        config: PropTypes.object.isRequired,
-        intl: intlShape.isRequired,
-        maxPostSize: PropTypes.number.isRequired,
-        shouldShowPreview: PropTypes.bool.isRequired,
-        useChannelMentions: PropTypes.bool.isRequired,
-
-        editingPost: PropTypes.shape({
-            post: PropTypes.object,
-            postId: PropTypes.string,
-            refocusId: PropTypes.string,
-            commentCount: PropTypes.number,
-            show: PropTypes.bool.isRequired,
-            title: PropTypes.string,
-            isRHS: PropTypes.bool,
-        }).isRequired,
-
-        actions: PropTypes.shape({
-            addMessageIntoHistory: PropTypes.func.isRequired,
-            editPost: PropTypes.func.isRequired,
-            hideEditPostModal: PropTypes.func.isRequired,
-            openModal: PropTypes.func.isRequired,
-            setShowPreview: PropTypes.func.isRequired,
-        }).isRequired,
-    }
-
-    constructor(props) {
+type OpenModal = {
+    ModalId: string;
+    dialogType: any;
+    dialogProps: {
+        post: {
+            message_source?: string;
+            message: string;
+            channel_id: string;
+            file_ids?: string[];
+        };
+        commentCount?: number;
+        isRHS?: boolean;
+    };
+};
+
+export type Props = {
+    canEditPost?: boolean;
+    canDeletePost?: boolean;
+    codeBlockOnCtrlEnter?: boolean;
+    ctrlSend?: boolean;
+    config: {
+        EnableEmojiPicker: string;
+        EnableGifPicker?: string;
+    };
+    intl: IntlShape;
+    maxPostSize: number;
+    shouldShowPreview: boolean;
+    useChannelMentions: boolean;
+    editingPost: {
+        post: {
+            message_source?: string;
+            message: string;
+            channel_id: string;
+            file_ids?: string[];
+        };
+        postId?: string;
+        refocusId?: string;
+        commentCount?: number;
+        show: boolean;
+        title?: string;
+        isRHS?: boolean;
+    };
+    actions: {
+        addMessageIntoHistory: (message: string) => void;
+        editPost: (input: any) => { data: any };
```

Comment:
Same here, any suggestions on how to type these?

Reply:
I believe there is a `Post` type in `packages/mattermost-redux` that should work here.

Reply:
Thanks! I added this but had to use `Pick` since the types don't match exactly.